Documentation/CRDs/ceph-client-crd.md

```diff
@@ -41,3 +41,121 @@ spec:
 ### Prerequisites
 
 This guide assumes you have created a Rook cluster as explained in the main [Quickstart guide](../Getting-Started/quickstart.md).
+
+## Example
+
+### Using a `CephClient` to create Ceph credentials
+
+To use `CephClient` to connect to a Ceph cluster:
+
+#### 1. Create the `CephClient` (`metadata.name` will be your Ceph client ID)
+
+Apply your CephClient resource YAML (see above for an example client config) with `kubectl`:
+
+```console
+kubectl apply -f example.cephclient.yaml
+```
+
+#### 2. Find the generated secret for the `CephClient`
+
+Once your `CephClient` has been processed by Rook (in the rook namespace, for example `rook-ceph`), it will be updated to include your secret:
+
+```console
+kubectl get cephclient exapmle -o jsonpath='{.status.info.secretName}' -n rook-ceph
+```
+
+#### 3. Extract Ceph cluster credentials from the generated secret
+
+Extract Ceph cluster credentials from the generated secret (note that the subkey will be your original client name):
+
+```console
+kubectl get secret rook-ceph-client-example -o jsonpath="{.data.example}" -n rook-ceph | base64 -d
+```
+
+The base64 encoded value that is returned *is* the password for your ceph client.
+
+#### 4. Determine the location of mons for the cluster
+
+To send writes to the cluster, you must retrieve the mons in use:
+
+```console
+kubectl get configmap rook-ceph-mon-endpoints -o jsonpath='{.data.data}' -n rook-ceph | sed 's/.=//g'`
+```
+
+This command *should* produce a line that looks somewhat like this:
+
+```
+XXX.XXX.XXX.XXX:6789,YYY.YYY.YYY.YYY:6789,ZZZ.ZZZ.ZZZ.ZZZ:6789
+```
+
+#### 4. (optional) Generate/create Ceph configuration files
+
+If you choose to generate files for Ceph to use you will need to generate the following files:
+
+- General configuration file (ex. `ceph.conf`)
+- Keyring file (ex. `ceph.keyring`)
+- File containing ceph arguments to use (ex. `ceph.args`)
+
+Examples of the files follow:
+
+`ceph.conf`
+```toml
+[global]
+mon_host=XXX.XXX.XXX.XXX:6789,YYY.YYY.YYY.YYY:6789,ZZZ.ZZZ.ZZZ.ZZZ:6789
+log file = /tmp/ceph-$pid.log
+```
+
+`ceph.keyring`
+```toml
+[client.example]
+  key = < key, decoded from k8s secret>
+  # The caps below are for a rbd workload -- you may need to edit/modify these capabilities for other workloads
+  # see https://docs.ceph.com/en/latest/cephfs/capabilities
+  caps mon = 'allow r'
+  caps osd = 'profile rbd pool=<your pool>, profile rb pool=<another pool>'
+```
+
+`ceph.args`
+```toml
+--id example
+```
```

Comment:
I'm having trouble understanding what the `ceph.args` file is. I assume this is meant to describe args you have to pass?

Reply:
hey yes, this is for the use with `libcephsqlite` -- one of the things you can specify is the args to use with Ceph -- it's not necessary in the general case, so I can move it down to the`libcephsqlite` section probably

Reply:
Actually thinking about it, I think it's necesary for `ceph` CLI use as well -- I guess I could change the command to `ceph status --id example` rather than specifying `CEPH_ARGS` via ENV.

What do you think?